### PR TITLE
kinds: Remove non-existent properties.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/theory/arith/kinds [hunk 1]
```diff
@@ -8,7 +8,7 @@ theory THEORY_ARITH ::CVC5::theory::arith::TheoryArith "theory/arith/theory_arit
 typechecker "theory/arith/theory_arith_type_rules.h"
 
 properties stable-infinite
-properties check propagate ppStaticLearn presolve notifyRestart
+properties propagate ppStaticLearn presolve notifyRestart
```
Comment (ajreynol) [inline src/theory/arith/kinds:11]:
```suggestion
properties ppStaticLearn presolve notifyRestart
```

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: src/theory/arith/kinds (hunk 1 above)
Comment (aniemetz) [reply src/theory/arith/kinds:11]: `propagate()` is defined in `theory_arith.h`, though?

Comment (ajreynol) [reply src/theory/arith/kinds:11]: Oh I see, it looked as though you were deleting the propagate property, I forgot this is not 100% standardized yet.

Nevermind :)

**Review by ajreynol — APPROVED** [review]

**Review by ajreynol — CHANGES_REQUESTED** [review]
Actually, this needs a bit more thought.  

I would actually recommend either:
(A) keeping the `check` property but disabling the warnings, or
(B) eliminate the `check` property (but fix it so that checks are called), and verify that it does not degrade performance.

The situation here is that some theories implement check(), others don't, however the code to check if the `check()` method exists is broken since this method is now implemented in the base class, not the derived ones.

**Review by ajreynol — APPROVED** [review]
LGTM

**Pull request merged**